src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs

```diff
@@ -0,0 +1,104 @@
+﻿using FluentMigrator;
+using Nop.Core.Domain.Blogs;
+using Nop.Core.Domain.Catalog;
+using Nop.Core.Domain.Customers;
+using Nop.Core.Domain.Forums;
+using Nop.Core.Domain.Logging;
+using Nop.Core.Domain.News;
+using Nop.Core.Domain.Orders;
+using Nop.Core.Domain.Polls;
+
+namespace Nop.Data.Migrations.UpgradeTo470
+{
+	[NopSchemaMigration("2023-11-09 00:00:02", "AddIndexesMigration for 4.70.0")]
+	public class AddIndexesMigration : ForwardOnlyMigration
+	{
+		/// <summary>
+		/// Collect the UP migration expressions
+		/// </summary>
+		public override void Up()
+		{
+			const string databaseType = "sqlserver";
+
+			IfDatabase(databaseType).Create.Index("IX_ActivityLog_CustomerId").OnTable(nameof(ActivityLog))
+				.OnColumn(nameof(ActivityLog.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_CustomerCustomerRoleMapping_CustomerId").OnTable(nameof(CustomerCustomerRoleMapping))
```

Comment:
For this table and its CustomerId field, you must use the NameCompatibilityManager, since they differ in the database and the code

Reply:
I have updated the code to use NameCompatibilityManager